src/Adapter/EntityTranslation/EntityTranslator.php

```diff
@@ -0,0 +1,238 @@
+                    $updateFields[] = '`' . bqSQL($fieldName) . '` = "' . pSQL($translatedField) . '"';
+                }
+            }
+
+            // Update table
+            if (!empty($updateWhere) && !empty($updateFields)) {
+                $updateWhere = implode(' AND ', $updateWhere);
+                $updateFields = implode(', ', $updateFields);
+
+                $sql = "UPDATE `$tableNameSql`
+                    SET $updateFields
+                    WHERE $updateWhere AND `id_lang` = $languageId"
+                    . $shopWhere
+                    . ' LIMIT 1';
+
+                $this->db->execute($sql);
+            }
+        }
+    }
+
+    /**
+     * Returns true if an id_shop field exists in database
+     *
+     * @param string $tableNameSql
+     *
+     * @return bool
+     *
+     * @throws \PrestaShopDatabaseException
+     */
+    protected function shopFieldExists(string $tableNameSql): bool
+    {
+        $columns = $this->db->executeS(
+            sprintf('SHOW COLUMNS FROM `%s`', $tableNameSql)
+        );
+
+        foreach ($columns as $column) {
+            if ($column['Field'] == 'id_shop') {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Retrieves the original wording via reverse dictionary search (aka "untranslation")
+     *
+     * @param array $data Database record
+     * @param string $fieldName Name of the field from $data to translate
+     *
+     * @return string "Untranslated" value
+     */
+    protected function getSourceString(array $data, string $fieldName): string
+    {
+        if (!array_key_exists($fieldName, $data)) {
+            throw new InvalidArgumentException(
+                sprintf(
+                    'Unable to reverse translate entity field "%s" because it\'s not defined in the provided database record',
+                    $fieldName
+                )
+            );
+        }
+
+        return $this->translator->getSourceString($data[$fieldName], $this->dataLang->getDomain());
+    }
+
+    /**
+     * Finds out the original wording and translates it
+     *
+     * @param array $data Database record
+     * @param string $fieldName Name of the field from $data to translate
+     *
+     * @return string Translated value
+     */
+    protected function doTranslate(array $data, string $fieldName): string
+    {
+        $untranslated = $this->getSourceString($data, $fieldName);
+
+        return $this->dataLang->getFieldValue($fieldName, $untranslated);
+    }
+
+    /**
+     * Builds the table name using the DataLang class as source
+     *
+     * @param DataLangCore $dataLang
+     *
+     * @return string The table name, including prefix
+     */
+    private function buildTableNameFromDataLang(DataLangCore $dataLang): string
+    {
+        $tableName = Inflector::tableize(get_class($dataLang));
+        if (substr($tableName, 0, strlen(_DB_PREFIX_)) !== _DB_PREFIX_) {
```

Comment:
Use `construct` parameter for `dbprefix` like in `DataLangFactory`

Reply:
Done!